src/genn/genn/code_generator/generateSynapseUpdate.cc

```diff
@@ -2,6 +2,7 @@
 
 // Standard C++ includes
 #include <string>
+#include <regex>
```

Comment:
it doesn't look like you're not actually using ``<regex>`` in this file so, if it's required, that suggests you need it include in "codeGenUtils.h" instead.